aiida/tools/graph/graph_traversers.py

```diff
@@ -0,0 +1,287 @@
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for functions to traverse AiiDA graphs."""
+
+from numpy import inf
+from aiida.common.links import GraphTraversalRules, LinkType
+
+
+def get_nodes_delete(starting_pks, **kwargs):
+    """
+    This function will return the set of all nodes that can be connected
+    to a list of initial nodes through any sequence of specified authorized
+    links and directions for deletion.
+
+    :type starting_pks: list or tuple
+    :param starting_pks: Contains the (valid) pks of the starting nodes (also accepts sets).
+    :param bool create_forward: will traverse CREATE links in the forward direction.
+    :param bool call_calc_forward: will traverse CALL_CALC links in the forward direction.
+    :param bool call_work_forward: will traverse CALL_WORK links in the forward direction.
+    """
+    traverse_links = parse_traversal_rules(GraphTraversalRules.DELETE, **kwargs)
+    function_output = traverse_graph(
+        starting_pks,
+        get_links=False,
+        links_forward=traverse_links['forward'],
+        links_backward=traverse_links['backward']
+    )
+    return function_output['nodes']
+
+
+def get_nodes_links_export(starting_pks, **kwargs):
+    """
+    This function will return the set of all nodes that can be connected
+    to a list of initial nodes through any sequence of specified authorized
+    links and directions for export. This will also return the links and
+    the traversal rules parsed.
+
+    :type starting_pks: list or tuple
+    :param starting_pks: Contains the (valid) pks of the starting nodes (also accepts sets)..
+    :param bool input_calc_forward: will traverse INPUT_CALC links in the forward direction.
+    :param bool create_backward: will traverse CREATE links in the backward direction.
+    :param bool return_backward: will traverse RETURN links in the backward direction.
+    :param bool input_work_forward: will traverse INPUT_WORK links in the forward direction.
+    :param bool call_calc_backward: will traverse CALL_CALC links in the backward direction.
+    :param bool call_work_backward: will traverse CALL_WORK links in the backward direction.
+    """
+    from aiida import orm
+
+    traverse_links = parse_traversal_rules(GraphTraversalRules.EXPORT, **kwargs)
+    traverse_output = traverse_graph(
+        starting_pks,
+        get_links=True,
+        links_forward=traverse_links['forward'],
+        links_backward=traverse_links['backward']
+    )
+
+    # The exporter wants a list of dictionaries for every link.
```

Comment:
I feel that the logic below this rather belongs in the export function itself. My idea was to have `get_nodes_delete` and `get_nodes_export` as very thin wrapper functions, that have the same interface and behavior and just prevent a caller having to import `GraphTraversalRules` and the `traverse_graph` and `parse_traversal_rules` methods separately.

Reply:
Same interface/behavior would include same output structure? i.e. have `get_nodes_delete` return the `rules_applied` and an empty list of "links" as well? Or maybe in that case then this functions should also include the `get_links` parameter and just pass it to the `traverse_graph`...but in that case would it even be ok to use different defaults for each?

Reply:
Yes, let's keep the interface consistent. The changes are good :+1: